src/Ng/Jobs/Catalog2PackageFixupJob.cs

```diff
@@ -0,0 +1,160 @@
+using System.Linq;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Auth;
+using Microsoft.WindowsAzure.Storage.Blob;
+using NuGet.Packaging.Core;
+using NuGet.Services.Configuration;
+using NuGet.Services.Metadata.Catalog;
+
+namespace Ng.Jobs
+{
+    public class Catalog2PackageFixupJob : NgJob
+    {
+        private const int MaximumPackageProcessingAttempts = 5;
+        private static readonly TimeSpan MaximumPackageProcessingTime = TimeSpan.FromMinutes(10);
+        private static readonly TimeSpan DonePollingInterval = TimeSpan.FromMinutes(1);
+
+        private CatalogIndexReader _catalogIndexReader;
+        private ICatalogIndexProcessor _processor;
+
+        public Catalog2PackageFixupJob(ITelemetryService telemetryService, ILoggerFactory loggerFactory)
+            : base(telemetryService, loggerFactory)
+        {
+            ThreadPool.SetMinThreads(MaxDegreeOfParallelism, 4);
+        }
+
+        public override string GetUsage()
+        {
+            return "Usage: ng catalog2packagefixup"
+                   + $"-{Arguments.Source} <catalog>"
+                   + $"-{Arguments.Verify} true/false"
+                   + $"-{Arguments.StorageAccountName} <azure-account>"
+                   + $"-{Arguments.StorageKeyValue} <azure-key> "
+                   + $"-{Arguments.StorageContainer} <azure-container>";
+        }
+
+        protected override void Init(IDictionary<string, string> arguments, CancellationToken cancellationToken)
+        {
+            var source = arguments.GetOrThrow<string>(Arguments.Source);
+            var storageAccount = arguments.GetOrThrow<string>(Arguments.StorageAccountName);
+            var storageKey = arguments.GetOrThrow<string>(Arguments.StorageKeyValue);
+            var storageContainer = arguments.GetOrThrow<string>(Arguments.StorageContainer);
+            var verify = arguments.GetOrDefault(Arguments.Verify, false);
+            var verbose = arguments.GetOrDefault(Arguments.Verbose, false);
+
+            // Prepare the catalog reader.
+            var httpMessageHandlerFactory = CommandHelpers.GetHttpMessageHandlerFactory(TelemetryService, verbose);
+            var collectorHttpClient = new CollectorHttpClient(httpMessageHandlerFactory());
+
+            _catalogIndexReader = new CatalogIndexReader(new Uri(source), collectorHttpClient, TelemetryService);
+
+            // Prepare the Azure Blob Storage client.
+            var credentials = new StorageCredentials(storageAccount, storageKey);
+            var account = new CloudStorageAccount(credentials, useHttps: true);
+
+            var container = account
+                .CreateCloudBlobClient()
+                .GetContainerReference(storageContainer);
+
+            // Prepare the HTTP Client
+            var httpClient = new HttpClient(new WebRequestHandler
```

Comment:
>HttpClient [](start = 33, length = 10)

Can we have a user agent please?

Reply:
Added. These downloads are going straight to Azure Blob Storage, so these shouldn't affect download counts.